modules/compute/src/main/java/org/apache/ignite/internal/compute/queue/QueueExecutionImpl.java

```diff
@@ -46,6 +48,8 @@ class QueueExecutionImpl<R> implements QueueExecution<R> {
 
     private final CompletableFuture<R> result = new CompletableFuture<>();
 
+    private final Lock changePriorityLock = new ReentrantLock();
```

Comment:
Using the lock is a good idea. But I think we can expand the lock to every place read/write to/from queue. In this case we can user ReadWriteLock and get rid of volatile both in `queueEntry` and `priority`.